src/main/java/com/example/ForexApp_v1/controller/TransactionController.java

```diff
@@ -4,77 +4,92 @@
-            return "index"; // wstawić stronę że błąd
+            return "singleTransaction";
         }
     }
+
     @PostMapping("/singleTransaction")
-    public String addToSingleTransaction(@ModelAttribute("transacDTO") TransacDTO transacDTO, Model model){
-      try {
-          Transac transac = transacService.addTransaction(transacDTO);
+        public String addToSingleTransaction(final @Valid @ModelAttribute("transacDTO") TransacDTO transacDTO, final Model model, final BindingResult bindingResult){
+        if (bindingResult.hasErrors()) {
+            return "singleTransaction";
+        }
+        try {
+          TransacDTO transac = transacService.addTransaction(transacDTO);
           model.addAttribute("transac", transac);
           return "singleTransaction";
-      }catch (Exception e){
+      }catch (ConstraintViolationException cve){
+            cve.printStackTrace();
+            return "singleTransaction";
+        }
+        catch (Exception e){
           e.printStackTrace();
-          return "index";  // wstawić stronę że błąd
-      }
+            return "singleTransaction";
+        }
     }
-    @GetMapping("/manyTransactions")
-    public String addToManyTransactions(Model model){
-
-        model.addAttribute("transacDTOList", transacDTOList);
-        model.addAttribute("newTransacDTO", new TransacDTO());
 
-        return "manyTransactions";
+    @GetMapping("/manyTransactions")
+    public String addToManyTransactions(final Model model){
+        try {
+            model.addAttribute("transacDTOList", transacDTOList);
+            model.addAttribute("newTransacDTO", new TransacDTO());
+            return "manyTransactions";
+        }catch (Exception e){
+            return "manyTransactions";
+        }
     }
 
     @PostMapping("/addTransactions")
-    public String showCreateFormTransactions(@ModelAttribute TransacDTO transacDTO){
+    public String showCreateFormTransactions(final @ModelAttribute(name = "transacDTO") TransacDTO transacDTO, final BindingResult bindingResult){
+        if (bindingResult.hasErrors()) {
+            return "manyTransactions";
+        }
         try {
             transacDTOList.add(transacDTO);
-
             return "redirect:/manyTransactions";
         }catch (Exception e){
             e.printStackTrace();
-            return "index";
+            return "manyTransactions";
         }
     }
+
     @PostMapping("/calculateTransactions")
-    public String showMultipleTransactions(Model model) {
+    public String showMultipleTransactions(final Model model) {
         try {
-            List<Transac> resultTransacList = transacService.addManyTransactions(transacDTOList);
+            List<TransacDTO> resultTransacList = transacService.addManyTransactions(transacDTOList);
             transacDTOList.clear();
-
             model.addAttribute("resultTransacList", resultTransacList);
             return "showResultTransactions";
         }catch (Exception e){
```

Comment:
Can you make the exception less generic?

Reply:
ok